### ISSUE TITLE
[Question] Height and width of an MRI?

### ISSUE THREAD

**Issue by tashrifbillah** [Deep-MI/FastSurfer#28]
`fslinfo` reports the following for my data:

```vim
data_type       INT16
dim1            176
dim2            256
dim3            256
dim4            1
datatype        4
pixdim1         1.000000
pixdim2         1.000000
pixdim3         1.000000
pixdim4         3.200000
cal_max         0.000000
cal_min         0.000000
file_type       NIFTI-1+
```

Then, what would be the height and width for [this](https://github.com/Deep-MI/FastSurfer/blob/master/FastSurferCNN/generate_hdf5.py#L166)?

**Comment by m-reuter** [comment]
Width and height are with respect to the selected slicing plane (axial (default), coronal or sagittal). Now which dimensions in your data correspond to what direction depends on the orientation in which the data is stored (which is not reported in the output above, but is part of the header, or can be determined via the affine matrix). Maybe use FreeSurfer mri_info to get more information (specifically the orientation). Most likely the first dimension is Left or Right.

**Comment by tashrifbillah** [comment]
I used [this](https://github.com/Deep-MI/FastSurfer/blob/c5e96778a237c7e21b1c525b8cf8cf1fc95cfdf2/FastSurferCNN/eval.py#L170) manually, couldn't have any luck with unequal `dim1, dim2, dim3`

**Comment by tashrifbillah** [comment]
Anyways, here is `mri_info` you asked for:

```vim
Volume information for sub-1001_ses-01_desc-Xc_T2w.nii.gz
          type: nii
    dimensions: 176 x 256 x 256
   voxel sizes: 1.000000, 1.000000, 1.000000
          type: FLOAT (3)
           fov: 176.000
           dof: 0
        xstart: -88.0, xend: 88.0
        ystart: -128.0, yend: 128.0
        zstart: -128.0, zend: 128.0
            TR: 3200.00 msec, TE: 0.00 msec, TI: 0.00 msec, flip angle: 0.00 degrees
       nframes: 1
       PhEncDir: UNKNOWN
       FieldStrength: 0.000000
ras xform present
    xform info: x_r =   1.0000, y_r =   0.0000, z_r =   0.0000, c_r =     0.5000
              : x_a =   0.0000, y_a =   1.0000, z_a =   0.0000, c_a =     0.5000
              : x_s =   0.0000, y_s =   0.0000, z_s =   1.0000, c_s =     0.5000
Orientation   : RAS
Primary Slice Direction: axial

voxel to ras transform:
                1.0000   0.0000   0.0000   -87.5000
                0.0000   1.0000   0.0000  -127.5000
                0.0000   0.0000   1.0000  -127.5000
                0.0000   0.0000   0.0000     1.0000

voxel-to-ras determinant 1

ras to voxel transform:
                1.0000   0.0000   0.0000    87.5000
                0.0000   1.0000   0.0000   127.5000
                0.0000   0.0000   1.0000   127.5000
                0.0000   0.0000   0.0000     1.0000


```

**Comment by LeHenschel** [comment]
Yes, for training FastSurfer we used conformed images which are 256 cubed. You should be able to conform them with the conform.py script or FreeSurfer's mri_convert -c if you want to.

**Comment by tashrifbillah** [comment]
> if you want to.

Oh I didn't seem to have a choice. If that is indeed the case, the `height` and `width` parameters should be removed from the line I referenced.

**Comment by isukrit** [comment]
Hi @m-reuter and @LeHenschel, any suggestions on how to un-conform the output (such that the new output is in the same space and dimensions as the original input image)?

**Comment by tashrifbillah** [comment]
Hello @isukrit , my name is Tashrif and I am an engineer, unrelated to this project, working with the kind of problem you mentioned. Although you haven't directed the question to me, I can help you with that.

> how to un-conform the output (such that the new output is in the same space and dimensions as the original input image)

Are you familiar with [ANTs](https://github.com/ANTsX/ANTs) tool for registration? `antsRegistrationSyNQuick.sh` and `antsApplyTransforms` are two commands that can be used in succession to do that. See [these two lines](https://github.com/pnlbwh/pnlNipype/blob/master/scripts/fs2dwi.py#L24) for example.

For your second comment, can you reference relevant lines so I can possibly second/refute your understanding?

**Comment by m-reuter** [comment]
Hi @isukrit , no need to install Ants. This FS command:
`mri_vol2vol --mov aparc+aseg...  --out aseginorigspace.nii.gz --targ originalinput.nii.gz --regheader --nearest`
should do the trick.

**Comment by isukrit** [comment]
@m-reuter and @tashrifbillah thanks for the info. I am using mri_vol2vol and this works as needed.
